toptal/picasso
# fix(ModalContent): extend useScrollableShades to support dynamic content

### Description

ModalContent has an embedded feature to display shades that should give user idea the content within modal is scrollable. Though it does not work if content of the modal changes from non-scrollable to scrollable and vice-versa.

### How to test

- Ensure scrolling shades on modal content appear as expected
- Ensure there's no performance degradation for modals that have content that's frequently updates, e.g. modal with a form

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| <video src="https://user-images.githubusercontent.com/903437/225295667-9f34c274-4c4a-42f7-a4c9-6d317e2eaed5.mp4" /> | <video src="https://user-images.githubusercontent.com/903437/225295742-30fdcebb-5986-44ee-956a-9928da7d1874.mp4" /> |

### Development checks


**Breaking change**


> All **_development checks_** should be done and set checked to pass the
> **GitHub Bot: TODOLess** action

<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run package:alpha-release` - Release alpha version
- `@toptal-anvil ping reviewers` - Ping FX team for review

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
